### ISSUE TITLE
Entering existing mail but wrong pw - auth response

### ISSUE THREAD

**Issue by rompic** [mattermost/platform#4321]
For feature requests, please see: http://www.mattermost.org/feature-requests/.  
For troubleshooting questions, please post on the forums: http://forum.mattermost.org/.

For issues with the product, please follow the steps at http://www.mattermost.org/filing-issues/ and post here in the following format: 
#### Summary

Issue in one concise sentence

Entering a correct mail but wrong pw provides "your password is incorrect" response. This is bad practice as It allows user/pw enumeration.
#### Steps to reproduce

How can we reproduce the issue (what version are you using?)

Go to Login screen
Enter correct Email
Enter wrong pw
Click on sign in
#### Expected behavior

Describe your issue in detail

A generic response should be provided. This is already the case if Email and pw don't exist.

See https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Email_address_as_a_User_ID
An application should respond with a generic error message regardless of whether the user ID or password was incorrect. It should also give no indication to the status of an existing account.
Incorrect Response Examples
"Login for User foo: invalid password"
"Login failed, invalid user ID"
"Login failed; account disabled"
"Login failed; this user is not active"
Correct Response Example
"Login failed; Invalid userID or password"
#### Observed behavior

What did you see happen? Please include relevant error messages and/or screenshots.
#### Possible fixes

If you can, link to the line of code that might be responsible for the problem

**Comment by jasonblais** [comment]
Thanks @rompic for the feedback!

Would you be interested sharing the feedback in our [UX Design Mattermost channel](https://pre-release.mattermost.com/core/channels/ux-design)?

This was a deliberate design decision, but feedback is always welcome. Posting in the UX Design channel gets you into more direct communication with the community and Mattermost UX team.

**Comment by rompic** [comment]
I get that this is should be a feature, but as discussed above It is a security risk.

**Comment by jasonblais** [comment]
Thanks @rompic for the feedback. 

Mattermost has gone through extensive penetration testing and security reviews and there's a long list of [security updates](http://about.mattermost.com/security-updates/) we've made thanks to our community and customers. 

In this particular case, it's a deliberate design decision we've made for the product in the context of overall system security. 

Our Mattermost tag line is "modern communication behind your firewall" and the expectation is that if security is a top concern--as it is for many users--then deployment happens in a private network. While running under existing policies and auth protocols of the private network, we prioritize a better user experience (telling a user what went wrong on login) ahead of preventing users on the private network from enumerating user accounts. 

When deploying Mattermost outside a firewall, an IT Admin chooses to expose their system to issues inherent in public cloud deployments. In this case, we generally follow the standards of leading online services. For example, Gmail--a SaaS service hosting private data from hundreds of millions of users--offers APIs that not only confirm whether an email account exists, it brings back the user's profile picture by default. 

Mattermost offers the option to be highly secure. It is the decision of admins on how they choose to deploy.
